test/parameter_inspection.jl

```diff
@@ -26,10 +27,20 @@ end
     @test params(m) == (
         t = (
             A = 6,
-            B = Opaque(5)
```

Comment:
Why are you changing the test for `params`? The idea is to leave `params` alone because changing it could be breaking. You should make new tests for your new methods (`deep_params`) and leave tests for the existing methods alone. (Only in the next major release can we remove `params`).